### ISSUE TITLE
Investigate how CORS works with servant

### ISSUE THREAD

**Issue by newlandsvalley** [newlandsvalley/tunebank#1]

**Comment by newlandsvalley** [comment]
I am experiencing difficulties with [servant-options](https://github.com/sordina/servant-options) at the moment.  I haven'I yet figured out the complex type system. I am seeing 2 problems:

* I need to annotate all my binary returned types to say that they also return JSON.  Presumably because they need to do so in the case of an OPTIONS response  when CORS support is queried by the browser. See https://github.com/sordina/servant-options/issues/1.

* Compilation fails for routes using BasicAuth:
```
No instance for (Servant.Foreign.Internal.GenerateList
                     NoContent
                     (Servant.Foreign.Internal.Foreign
                        NoContent
                        (BasicAuth "tunebank-realm" UserName
                         :> (QueryParam "page" Int
                             :> (QueryParam "size" Int :> Get '[JSON] UserList)))))
``` 
See https://github.com/sordina/servant-options/issues/3.

**Comment by newlandsvalley** [comment]
OK - abandoning servant-options and am just using simpleCors.  This means we won't have a fine-grained policy but I don't really need one anyway. Fixed by 2524d55e77d26729f750495ba4eba2f57bb2edbf.

**Comment by newlandsvalley** [comment]
`simpleCors` has a policy which excludes BasicAuth.  We need to extend it by adding the Authorisation header to the acceptable request headers list.  Fixed by b4387340df18dc3dca408948f7588b155d012989.

**Comment by newlandsvalley** [comment]
Ditto for allowing the DELETE method - fixed by 5eec8ffdf12437c974442955522b68240bc7aa59.
